### PR TITLE
Bump golang to 1.18

### PR DESCRIPTION
**Pull request by e0ne**
Signed-off-by: Ivan Kolodiazhny <ikolodiazhny@nvidia.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Package related
 BINARY_NAME=ib-sriov
 PACKAGE=ib-sriov-cni
-ORG_PATH=github.com/Mellanox
+ORG_PATH=github.com/k8snetworkplumbingwg
```
Comment (adrianchiris) [inline Makefile:4]: can you also fix path at L27 (TAG)

File: Makefile [hunk 2]
```diff
@@ -67,22 +67,20 @@ $(BUILDDIR)/$(BINARY_NAME): $(GOFILES) | $(BUILDDIR)
 	@cd $(BASE)/cmd/$(PACKAGE) && CGO_ENABLED=0 $(GO) build -o $(BUILDDIR)/$(BINARY_NAME) -tags no_openssl -ldflags $(LDFLAGS) -v
 
 # Tools
-$(GOLANGCI_LINT): | $(BASE) ; $(info  building golangci-lint...)
-	$Q curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOBIN) $(GOLANGCI_LINT_VER)
+$(GOLANGCI_LINT): ; $(info  installing golangci-lint...)
+#	$(call go-install-tool,$(GOLANGCI_LINT),github.com/golangci/golangci-lint/cmd/golangci-lint@$(GOLANGCI_LINT_VER))
```
Comment (adrianchiris) [inline Makefile:71]: nit: remove this line

File: Makefile [hunk 3]
```diff
@@ -67,22 +67,20 @@ $(BUILDDIR)/$(BINARY_NAME): $(GOFILES) | $(BUILDDIR)
...
+	@go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(GOLANGCI_LINT_VER)
```
Comment (adrianchiris) [inline Makefile:72]: use $Q like in line 77 ?

File: go.mod [hunk 4]
```diff
@@ -1,19 +1,35 @@
 module github.com/Mellanox/ib-sriov-cni
 
-go 1.13
+go 1.17
```
Comment (adrianchiris) [inline go.mod:3]: you can use 1.18 here IMO

**Review by adrianchiris — COMMENTED** [review]
Thanks for the PR @e0ne  !

mind tidying up also Dockerfiles ? :)

1. remove Dockerfile.rhel7 as i dont believe its needed, plus it uses old go 1.13
2. set base image to golang 1.18 in Dockerfile and Dockerfile.ppc64le in build step

**Comment by adrianchiris** [comment]
@e0ne is this PR still relevant ? maybe we should bump to 1.19 :)

**Comment by e0ne** [comment]
> @e0ne is this PR still relevant ? maybe we should bump to 1.19 :)

I'll update my PR to use golang v1.19

**Comment by e0ne** [comment]
Closed in flavor of #70

**Pull request closed without merging**